internals-js/src/operations.ts

```diff
@@ -1598,9 +1598,18 @@ function addOneToKeyedUpdates(keyedUpdates: MultiMap<string, SelectionUpdate>, s
   }
 }
 
+function maybeRebaseOnSchema(toRebase: CompositeType, schema: Schema): CompositeType {
```

Comment:
This behavior is slightly different than I would expect. If the type is undefined in the new schema, why return the original type rather than, say, undefined?

Reply:
I guess that was me being overly cautious and defaulting to what was essentially the prior behaviour before this method was added if something unexpected happened. But truly, we should always find the type or we're having a bug, so I ended up changing this to an assertion. Probably better to get a signal early if we introduce/have such a bug (returning `undefined` would push the problem to the caller, which in that case wouldn't have much better it can do than failing, so having an `assertion` directly kept the code simpler; this is all non-exported methods after all).